Fix : Compiler eslint suppression silent skip 35105

manNomi opened the pull request:
## Summary

This PR fixes \#35105 where `eslint-disable` comments cause the React Compiler to skip entire functions, preventing important diagnostics like `incompatible-library` warnings from being shown.

**Motivation:** When developers add unrelated ESLint suppressions (e.g., for `exhaustive-deps`), the compiler would silently skip analyzing the entire function. This meant critical warnings about incompatible APIs would never reach the developer, leading to silent performance issues in production.

**Solution:** Implemented the "shorter term fix" suggested by @josephsavona - in ESLint mode (`noEmit: true`), suppressions are now logged as diagnostics but compilation **continues** to detect other issues. Build mode behavior is unchanged for backward compatibility.

### Real-World Testing

| As is | To be|
| :-----: | :-----: |
| <img width="596" height="452" alt="스크린샷 2025-11-13 오전 9 25 18" src="https://github.com/user-attachments/assets/6fdc371d-2a3b-4b65-a54a-2dac4df1c4c5" /> | <img width="573" height="478" alt="스크린샷 2025-11-13 오전 9 23 36" src="https://github.com/user-attachments/assets/8398e455-a173-49c7-865d-58675ec7d1a9" />
   |

josephsavona commented:
Thanks for the contribution. Your description says you "Created and ran a validation script that:" but...that script is not here. It is also unnecessary, since we have a testing framework. You (or the LLM you're clearly using) needs to actually run our tests :-)

josephsavona reviewed (changes requested):
see above

manNomi committed:
Keep ESLint suppressions from hiding diagnostics

manNomi committed:
snap update

manNomi commented:
I sincerely apologize again, both for the initial confusion and for wasting your valuable time.

As a non-native English speaker, I relied on an LLM to help draft the PR description. I failed to properly verify its output and it was a careless mistake on my part. I am truly sorry that this caused frustration when my only intention was to be helpful.

I have now done the following:
1.  Removed the incorrect part from the PR description.
2.  Run all local tests (`yarn snap`) which are now passing.
3.  Pushed the updated fixtures (`yarn snap --update`) .

I believe this now correctly addresses all the issues you pointed out.

However, if this contribution is still problematic or causing any inconvenience, **please let me know. I am more than willing to close this PR immediately so as not to cause any further trouble.**

Thank you so much for your patience and guidance.

manNomi committed:
Merge branch 'main' into fix/eslint-suppression-silent-skip-35105

manNomi committed:
Merge branch 'main' into fix/eslint-suppression-silent-skip-35105

manNomi committed:
Merge branch 'main' into fix/eslint-suppression-silent-skip-35105

manNomi committed:
Merge branch 'main' into fix/eslint-suppression-silent-skip-35105

manNomi committed:
Merge branch 'main' into fix/eslint-suppression-silent-skip-35105

manNomi commented:
bump
